src/Microsoft.AspNetCore.Mvc.Core/RequireHttpsAttribute.cs

```diff
@@ -43,6 +47,13 @@ protected virtual void HandleNonHttpsRequest(AuthorizationFilterContext filterCo
                     request.Path.ToUriComponent(),
                     request.QueryString.ToUriComponent());
 
+                // use a specific SSL port if it is provided in the options,
+                // clear it otherwise to use an empty default.
+                var uri = new UriBuilder(newUrl);
```

Comment:
After talking to @Tratcher we need to expose the port and domain separately in `HostString`. And be able to construct a new one from domain and port. I will then remove `UriBuilder` usage. https://github.com/aspnet/HttpAbstractions/issues/567